src/util.rs

```diff
@@ -133,6 +133,36 @@ pub fn is_local(ip_addr: &IpAddr, interfaces: &[Interface]) -> bool {
     false
 }
 
+fn ip(addr: &net::SocketAddr) -> IpAddr {
+    match *addr {
+        net::SocketAddr::V4(ref addr) => IpAddr::V4(*addr.ip()),
+        net::SocketAddr::V6(ref addr) => IpAddr::V6(*addr.ip()),
+    }
+}
+
+/// If the endpoint IP address is unspecified return a copy of the endpoint with the IP address
+/// set to the loopback address. Otherwise return a copy of the endpoint.
```

Comment:
This should be config test AFAIK as it's only used for the test function. All in all I feel it needs removed though.